bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php

```diff
@@ -216,19 +221,43 @@ public function treeGetChildsByIdAction(Request $request, EventDispatcherInterfa
         return $this->adminJson($objects);
     }
 
+    /**
+     * @param array $objects
+     * @param string $sortBy
+     * @param string $childrenSortOrder
+     * @return array
+     */
+    private function sortChildrenNaturally(array $objects, string $sortBy, string $childrenSortOrder): array
```

Comment:
Hey @mehlichmeyer I tested your changes and found that getChildren() API call and tree hierarchy still have different children order, reason being that `sortChildrenNaturally()` is never called in `getChildren()`. I would suggest to move this method to `DataObject\Service` and then we call it in getChildren() also.
```php
public static function sortChildrenNaturally(array $objects, string $sortBy, string $childrenSortOrder): array
    {

        usort(
            $objects,
            function ($a, $b) use ($sortBy) {
                return strnatcasecmp($a->get($sortBy), $b->get('o_'. $sortBy));
            }
        );

        return $childrenSortOrder === 'ASC' ? $objects : array_reverse($objects);
    }
```

Reply:
Hi @dvesh3 could you check again? :)

Reply:
@mehlichmeyer please check my last comment. Thanks!